### PR TITLE
add fail2ban support

### PR DESCRIPTION
**Pull request by VVelox**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

This adds Fail2ban support.

This also depends on the item below.

https://github.com/librenms/librenms-agent/pull/84

### CODE REVIEW

**Comment by VVelox** [comment]
blarg!

After lots of mucking around with git, I've fixed it so it will now properly display under the apps area.

File: doc/Extensions/Applications.md [hunk 1]
```diff
@@ -487,3 +488,21 @@ extend phpfpmsp /etc/snmp/phpfpm-sp
 7. On the device page in Librenms, edit your host and check `PHP-FPM` under the Applications tab.
 
 It is worth noting that this only monitors a single pool. If you want to monitor multiple pools, this won't do it.
+
+#### Fail2ban
+
+##### SNMP Extend
+
+1: Copy the shell script, fail2ban, to the desired host (the host must be added to LibreNMS devices) (wget https://github.com/librenms/librenms-agent/raw/master/snmp/fail2ban -O /etc/snmp/phpfpm-sp)
```
Comment (murrant) [inline doc/Extensions/Applications.md]: this is the wrong target file :0 should be /etc/snmp/fail2ban

**Comment by VVelox** [comment]
Awesome. Corrected there.

**Comment by VVelox** [comment]
Updated it to work with the newest snmpd extend.

It now creates RRDs for each jail. Still needs to update the graphing for it.

**Comment by laf** [comment]
Let us know when you're finished with this :)

**Comment by VVelox** [comment]
All good now. :)

**Comment by VVelox** [comment]
Just pondering the idea to add a third item that computes the difference between banned and firewalled and displaces it. Ya'lls thoughts?

Just thinking that may be useful instead of eyeballing the difference. Not sure how general purpose useful that would be though.

File: html/pages/device/apps/fail2ban.inc.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    </div>
+    <div class="panel-body">
+    <div class="row">';
+    include 'includes/print-graphrow.inc.php';
+    echo '</div>';
+    echo '</div>';
+    echo '</div>';
+}
+
+$baseName=$config['rrd_dir'].'/'.$device['hostname'].'/app-fail2ban-'.$app['app_id'].'-';
```
Comment (murrant) [inline html/pages/device/apps/fail2ban.inc.php]: Please use rrdname() here.

Comment (VVelox) [inline html/pages/device/apps/fail2ban.inc.php]: Sweet. Replaced with...
```php
$baseName=rrd_name($device['hostname'], 'app-fail2ban-', $app['app_id'].'-');
```
And it is working nicely.

**Comment by laf** [comment]
I'm ok with this.

**Comment by murrant** [comment]
I updated this so it works again (recent change in polling).  Polling is working.

I don't see any graphs for this on the graph page (not even blank ones).  Can you check this on your install?

**Comment by VVelox** [comment]
Did a clone of master and no graphs are rendering now at all, not just this.

**Comment by VVelox** [comment]
hmm... had to set the path for rrdtool for it... looking now

**Comment by VVelox** [comment]
hmm... what ever changed has also broken the PHP-FPM app

**Comment by VVelox** [comment]
DHCP Stats is also broken.

**Comment by VVelox** [comment]
Hmm... Everything using 'includes/graphs/generic_v3_multiline_float.inc.php' is broken.

**Comment by VVelox** [comment]
hmm... appears that any app graph with more than one DS is broken here for me

**Comment by VVelox** [comment]
Implemented the fixes implemented in the pull below for other graphs.

https://github.com/librenms/librenms/pull/5994

**Pull request merged**